core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/test/PlanBuilder.java

```diff
@@ -499,7 +499,7 @@ protected AggregationNode build()
         {
             checkState(groupingSets != null, "No grouping sets defined; use globalGrouping/groupingKeys method");
             return new AggregationNode(
-                    nodeId.orElse(idAllocator.getNextId()),
+                    nodeId.orElseGet(idAllocator::getNextId),
```

Comment:
(X) Object creation and minor calculations

Reply:
not side-effect free